api/configuration_default.go

```diff
@@ -27,6 +27,15 @@ func NewAPIConfig(params ConfigParams) *Configuration {
 	apiConfig.Scheme = params.Host.Scheme
 	apiConfig.UserAgent = params.UserAgent
 	apiConfig.HTTPClient = &http.Client{Transport: clientTransport}
+
+	if params.Host.Path != "" {
+		// initialize api server configurations with path of host  url
+		apiConfig.Servers = ServerConfigurations{{URL: params.Host.Path}}
+		for key := range apiConfig.OperationServers {
+			apiConfig.OperationServers[key] = ServerConfigurations{{URL: params.Host.Path}}
```

Comment:
Overall, this looks relatively straightforward and like it will address the issue. 

My one reservation is that it completely overrides the existing `apiConfig.OperationServers[key]` value. We don't lose any valuable information today (just a non-useful description), but I'm concerned this could introduce subtle issues in the future if more valuable information is already populated in the `ServerConfigurations`.

Would it be possible to set the `URL` value without overwriting everything else?

Reply:
@gwossum Just to handle the path part, you can take a look at #530 which do a light prepend on constructing the final request.